gluten-core/src/main/scala/io/glutenproject/softaffinity/SoftAffinityManager.scala

```diff
@@ -47,6 +53,30 @@ abstract class AffinityManager extends LogLevelUtil with Logging {
 
   lazy val logLevel: String = GlutenConfig.getConf.softAffinityLogLevel
 
+  lazy val detectDuplicateReading = true
+
+  lazy val maxDuplicateReadingRecords =
+    GlutenConfig.GLUTEN_SOFT_AFFINITY_MAX_DUPLICATE_READING_RECORDS_DEFAULT_VALUE
+
+  val totalDuplicateReadingRecords = new AtomicInteger(0)
+  val totalSoftAffinitySetRecords = new AtomicInteger(0)
+  val totalSoftAffinityMissRecords = new AtomicInteger(0)
+  val totalSoftAffinityEffiencyRecords = new AtomicInteger(0)
+
+  val querySoftAffinityDetectInfo = new ConcurrentHashMap[Long, Int]()
+  val querySoftAffinityDetectMissInfo = new ConcurrentHashMap[Long, Int]()
+  val querySoftAffinityDetectResultInfo = new ConcurrentHashMap[Long, Int]()
+
+  // rdd id -> patition id, file path, start, length
+  val rddPartitionInfoMap = new ConcurrentHashMap[Int, Array[(Int, String, Long, Long)]]()
```

Comment:
With large spark job this could be very big? especially the file path is also incuded

Reply:
We add config `GLUTEN_SOFT_AFFINITY_MAX_DUPLICATE_READING_RECORDS` to control the map size. If the records in map > this threshold, then recording logic would be skipped.

Reply:
can we consider to use `LoadingCache` to save these records, and then set the `maximumSize` of the `LoadingCache` to control the max records count, while it will use the  LRU policy to remove unused key.

Reply:
Thanks, I updated `duplicateReadingInfos` with `LoadingCache`. For others maps they are used to store temp informations and would be clear after stages ending, so I keep them as before.